lib/src/sip_ua_helper.dart

```diff
@@ -93,6 +94,14 @@ class SIPUAHelper extends EventManager {
     if (_ua != null && _ua!.isConnected()) {
       Map<String, dynamic> options = buildCallOptions(voiceonly);
       if (customOptions != null) {
+        if (customOptions.containsKey('from_uri')) {
+          if (!customOptions['from_uri']
+              .contains(RegExp(r'^sip:', caseSensitive: false))) {
+            customOptions['from_uri'] =
+                '${DartSIP_C.SIP}:${customOptions['from_uri']}';
+          }
+          customOptions['from_uri'] = URI.parse(customOptions['from_uri']);
+        }
```

Comment:
Can you refactor this whole function? It's doing too much and too many responsibilities.

I'd start by taking out the parent if-statement

```
  if (_ua == null || !_ua!.isConnected()) {
    logger.e('Not connected, you will need to register.', null, StackTraceNJ());
    return false;
  }
```